QModManager/HarmonyPatches/QuitToDesktop.cs

```diff
@@ -0,0 +1,102 @@
+            QTD.quitButton.name = "QuitToDesktop Button";
+            QTD.quitButton.onClick.RemoveAllListeners();
+            QTD.quitButton.onClick.AddListener(() => QTD.QuitDesktopSubscreen(__instance));
+
+            var confirmationPrefab = __instance.transform.Find("QuitConfirmation").gameObject;
+            QTD.quitConfirmation = GameObject.Instantiate(confirmationPrefab, __instance.transform);
+            QTD.quitConfirmation.name = "QuitToDesktop Confirmation";
+            QTD.quitConfirmation.GetComponentsInChildren<Button>()[1].onClick.RemoveAllListeners();
+            QTD.quitConfirmation.GetComponentsInChildren<Button>()[1].onClick.AddListener(() => __instance.QuitGame(true));
+
+            var confirmationWithSaveWarningPrefab = __instance.transform.Find("QuitConfirmationWithSaveWarning").gameObject;
+            QTD.quitConfirmation2 = GameObject.Instantiate(confirmationWithSaveWarningPrefab, __instance.transform);
+            QTD.quitConfirmation2.name = "QuitToDesktop ConfirmationWithSaveWarning";
+            QTD.quitConfirmation2.GetComponentsInChildren<Button>()[1].onClick.RemoveAllListeners();
+            QTD.quitConfirmation2.GetComponentsInChildren<Button>()[1].onClick.AddListener(() => __instance.QuitGame(true));
+        }
+    }
+
+    [HarmonyPatch(typeof(IngameMenu), nameof(IngameMenu.Open))]
+    internal static class IngameMenu_Open_Patch
+    {
+        // This patch disables the QTD button in hardcore mode and changes the names of the two quit buttons based on game mode
+
+        [HarmonyPostfix]
+        internal static void Postfix(IngameMenu __instance)
+        {
+            __instance.quitToMainMenuText.text = $"{(GameModeUtils.IsPermadeath() ? "Save & " : "")}Quit to Main Menu";
+            QTD.quitButton.interactable = !GameModeUtils.IsPermadeath();
+            QTD.quitButton.GetComponentsInChildren<Text>().Do(t => t.text = "Quit to Desktop");
+        }
+    }
+
+    /*
+    [HarmonyPatch(typeof(IngameMenu), nameof(IngameMenu.QuitGame))]
+    internal static class IngameMenu_QuitGame
+    {
+        // This is a pass-through patch
+
+        [HarmonyPrefix]
+        internal static bool Prefix(IngameMenu __instance, bool quitToDesktop)
+        {
+            __instance.StartCoroutine(IngameMenu_QuitGameAsync.Passthrough(__instance.QuitGameAsync(quitToDesktop), quitToDesktop));
+            return false;
+        }
+    }
+
+    internal static class IngameMenu_QuitGameAsync
+    {
+        // This patch makes QTD work in hardcore mode
+
+        internal static IEnumerator Passthrough(IEnumerator original, bool quitToDesktop)
+        {
+            int n = 0;
+            while (original.MoveNext())
+            {
+                yield return original.Current;
+
+                if (quitToDesktop && original.Current is IEnumerator)
+                    yield return new WaitForSecondsRealtime(2);
+
+                n++;
+            }
+        }
+    }
+    */
```

Comment:
If this is no longer needed, remove the commented code.

Reply:
Is this commented code necessary?

Reply:
It's my attempt to make QTD work in hardcore. It's 99% successful. I'll still work on it before 3.1 releases